cassandra/cqlengine/connection.py

```diff
@@ -281,8 +329,6 @@ def set_session(s):
     if conn.session:
         log.warning("configuring new default connection for cqlengine when one was already set")
 
-    if s.row_factory is not dict_factory:
-        raise CQLEngineException("Failed to initialize: 'Session.row_factory' must be 'dict_factory'.")
```

Comment:
Moved this into the Connection class so it is called consistently. If using a different way to initialize the connection (e.g., `Connection.from_session()`), this check wasn't happening.